sdk/openai/Azure.AI.OpenAI/src/Generated/CompletionsLogProbabilityModel.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.Collections.Generic;
+using System.Linq;
+using Azure.Core;
+
+namespace Azure.AI.OpenAI
+{
+    /// <summary> Representation of a log probabilities model for a completions generation. </summary>
+    public partial class CompletionsLogProbabilityModel
+    {
+        /// <summary> Initializes a new instance of CompletionsLogProbabilityModel. </summary>
+        internal CompletionsLogProbabilityModel()
+        {
+            Tokens = new ChangeTrackingList<string>();
+            TokenLogProbabilities = new ChangeTrackingList<float>();
+            TopLogProbabilities = new ChangeTrackingList<IDictionary<string, float>>();
+            TextOffsets = new ChangeTrackingList<int>();
+        }
+
+        /// <summary> Initializes a new instance of CompletionsLogProbabilityModel. </summary>
+        /// <param name="tokens"> The textual forms of tokens evaluated in this probability model. </param>
+        /// <param name="tokenLogProbabilities"> A collection of log probability values for the tokens in this completions data. </param>
+        /// <param name="topLogProbabilities"> A mapping of tokens to maximum log probability values in this completions data. </param>
+        /// <param name="textOffsets"> The text offsets associated with tokens in this completions data. </param>
+        internal CompletionsLogProbabilityModel(IReadOnlyList<string> tokens, IReadOnlyList<float> tokenLogProbabilities, IReadOnlyList<IDictionary<string, float>> topLogProbabilities, IReadOnlyList<int> textOffsets)
+        {
+            Tokens = tokens.ToList();
+            TokenLogProbabilities = tokenLogProbabilities.ToList();
+            TopLogProbabilities = topLogProbabilities.ToList();
+            TextOffsets = textOffsets.ToList();
+        }
+
+        /// <summary> The textual forms of tokens evaluated in this probability model. </summary>
+        public IReadOnlyList<string> Tokens { get; }
+        /// <summary> A collection of log probability values for the tokens in this completions data. </summary>
+        public IReadOnlyList<float> TokenLogProbabilities { get; }
+        /// <summary> A mapping of tokens to maximum log probability values in this completions data. </summary>
+        public IReadOnlyList<IDictionary<string, float>> TopLogProbabilities { get; }
+        /// <summary> The text offsets associated with tokens in this completions data. </summary>
+        public IReadOnlyList<int> TextOffsets { get; }
```

Comment:
I believe it's just telling you the substring indices of what's in `tokens`. I don't see any particular utility in it given you can just look at the lengths in `tokens`.

E.g. if I have a partial `tokens` array of
```
"tokens": [
  "Are",
  " bananas",
  " especially",
  " radioactive",
```

I end up with `TextOffsets` (partial) of
```
"text_offset": [0, 3, 11, 22, 34],
```

Think we should just remove this from the projection?

Reply:
Gotcha. Thank you for the explanation! This doesn't need to be considered as part of this PR, but it got me thinking because I've seen something similar in other Azure services like Text Analytics and Form Recognizer. The interesting thing about these offsets is that the text encoding that is used can affect them and the expectations on how to use them, especially when using emoji and multilingual text. For context, .NET uses UTF-16 by default to encode the text in a `string`, so when using methods like `Substring()`, the expectation is that the offsets and lengths are adjusted for UTF-16.

Here's more info about the problem and the approach taken by the Azure Cognitive Service for Language: https://learn.microsoft.com/en-us/azure/cognitive-services/language-service/concepts/multilingual-emoji-support

This might not be particularly important in this case, since, like you say, the utility of these `TextOffsets` is questionable.